h2o-core/src/main/java/hex/Model.java

```diff
@@ -873,10 +879,9 @@ public Model(Key<M> selfKey, P parms, O output) {
     _output = output;  // Output won't be set if we're assert output != null;
     if (_output != null)
       _output.startClock();
-    _dist = isSupervised() && _output.nclasses() == 1 ? DistributionFactory.getDistribution(_parms) : null;
+    _dist = isSupervised() && _output.nclasses() == 1 ? DistributionFactory.getDistribution(parms) : null;
```

Comment:
why this change? once the field is set, we should keep working just with the field reference - if anyone clones or modifies the object it breaks your code in a bad way

Reply:
It was a mistake when I tried something to change here. Good catch.